src/Avalonia.Interactivity/EventRoute.cs

```diff
@@ -0,0 +1,200 @@
+﻿using System;
+using Avalonia.Collections.Pooled;
+
+namespace Avalonia.Interactivity
+{
+    /// <summary>
+    /// Holds the route for a routed event and supports raising an event on that route.
+    /// </summary>
+    public class EventRoute : IDisposable
```

Comment:
This could be a `struct` as well. Perhaps even `readonly struct` but we would need to remove that set to null in `Dispose`.

Reply:
After our discussion about ABI - we might want to leave this as a class then.